src/puppetlabs/puppetdb/query.clj

```diff
@@ -422,21 +434,23 @@
           (:where %)]}
   (match [path]
          ["tag"]
-         {:where (sql-regexp-array-match "catalog_resources" "catalog_resources" "tags")
+         {:where (.s (sql-regexp-array-match "catalog_resources"
+                                             "catalog_resources"
+                                             "tags"))
           :params [value]}
 
          ;; node join.
          ["certname"]
-         {:where  (sql-regexp-match "catalogs.certname")
+         {:where (legacy-sql-regexp-match "catalogs.certname")
```

Comment:
oh I get it, so we were returning honeysql here, but never converting the honeysql into a string?